fennel/datasets/test_dataset.py

```diff
@@ -2580,3 +2580,67 @@ def pipeline2(cls, a: Dataset, b: Dataset):
     sync_request = view._get_sync_request_proto(tier="prod")
     pipelines = sync_request.pipelines
     assert len(pipelines) == 1
+
+
+def test_dataset_with_str_window_aggregate():
+    @meta(owner="test@test.com")
+    @dataset
+    class UserAggregatesDataset:
+        gender: str = field(key=True)
+        timestamp: datetime = field(timestamp=True)
+        count: int
+        avg_age: float
+
+        @pipeline(version=1)
+        @inputs(UserInfoDataset)
+        def create_aggregated_dataset(cls, user_info: Dataset):
+            return user_info.groupby("gender").aggregate(
+                [
+                    Count(
+                        window="forever",
+                        into_field=str(cls.count),
+                    ),
+                    Average(
+                        of="age",
+                        window=Window("forever"),
+                        into_field=str(cls.avg_age),
+                    ),
+                ]
+            )
```

Comment:
Try to fit aggregates in one line each by removing `[`

Reply:
got it